pandas/io/parsers.py

```diff
@@ -3791,3 +3735,93 @@ def _check_defaults_read(
         kwds["engine_specified"] = False
 
     return kwds
+
+
+def _merge_with_dialect_properties(
```

Comment:
reverse these (always put kwds things last).

Reply:
rename kwds -> defaults

so similar to the _check_defaults_read

also not averse to renaming _check_defaults_read to something better.

Reply:
I have done it.
Also, I found the issue with the ValueError message, when dialect is incorrect.
Originally, kwds['dialect'], but I decided to use dialect.__name__ for better readability, which is not accepted by mypy (throws error ``"Dialect" has no attribute "__name__"  [attr-defined]``).
So, now I use string representation of dialect instance.